src/amo/components/VPNPromoBanner/index.js

```diff
@@ -70,27 +91,40 @@ export class VPNPromoBannerBase extends React.Component<InternalProps> {
   }
 
   onButtonClick: () => void = () => {
-    this.props._tracking.sendEvent({
+    const { _tracking, _window } = this.props;
+
+    const impressionCount = getImpressionCount(_window);
+    _tracking.sendEvent({
       action: VPN_PROMO_CLICK_ACTION,
       category: VPN_PROMO_CATEGORY,
+      label: String(impressionCount),
     });
+    clearImpressionCount(_window);
   };
 
   onDismiss: () => void = () => {
-    this.props._tracking.sendEvent({
+    const { _tracking, _window } = this.props;
+
+    const impressionCount = getImpressionCount(_window);
+    _tracking.sendEvent({
       action: VPN_PROMO_DISMISS_ACTION,
       category: VPN_PROMO_CATEGORY,
+      label: String(impressionCount),
     });
+    clearImpressionCount(_window);
   };
```

Comment:
Looks like both `onButtonClick` and `onDismiss` are very similar, maybe we could have a shared method that would take a single `action` argument.

Reply:
The issue is that we're using the `onClick={this.onButtonClick}` syntax, so I'm not sure if we can pass an argument to the function. We can interrogate the target of the event, but it seems like adding that wouldn't really make a single function much less code than having the two separate ones. Is there a trick to easily pass an argument using the `onClick={this.onButtonClick}` syntax?

Reply:
Hmm, I think I see some solutions. I'll give it a try.

Reply:
I think the best solution that doesn't have a performance impact on render (e.g., using `onClick={() => this.onButtonClick(action)}`), and doesn't involve interrogating the event, is to just have two functions which each call the common function. It's nearly as much code as having both function separate, but at least the logic is only implemented once. I've updated the patch with that change.